src/map.js

```diff
@@ -0,0 +1,233 @@
+import React from "react";
+import L from "leaflet";
+import borderData from "./border.js";
+import leafletPip from "@mapbox/leaflet-pip";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      markerPosition: { lat: 44.0286, lng: -72.7317 },
+      oldMarkerPosition: { oldLat: 44.0886, oldLng: -72.7317 },
+      minimumZoom: 18,
+      maximumZoom: 18,
+      defaultZoom: 18,
+      gameStarted: false,
+      borderData: L.geoJSON(borderData),
+      currentPoint: { lat: 44.0286, lng: -72.7317 },
+      randomCord: { lat: 44.0286, lng: -72.7317 }
+    };
+
+    // binding the functions to the state
+    this.moveMapNorth = this.moveMapNorth.bind(this);
+    this.moveMapSouth = this.moveMapSouth.bind(this);
+    this.moveMapEast = this.moveMapEast.bind(this);
+    this.moveMapWest = this.moveMapWest.bind(this);
+
+    this.startGame = this.startGame.bind(this);
+    this.quitGame = this.quitGame.bind(this);
+
+    this.randomCord = this.randomCord.bind(this);
+
+    this.randomLat = this.randomLat.bind(this);
+    this.randomLong = this.randomLong.bind(this);
+  }
+
+  // startGame function generates a random point within the declared boundaires 
+  startGame() {
+    this.setState({
+      gameStarted: true
+    });
+    if (this.state.gameStarted) {
+      this.randomCord();
+    }
+  }
+
+  // formula for generating the random coordinates 
+  randomCord() {
+    this.setState({
+      gameStarted: true
+    });
+    let randomLat = this.randomLat();
+    console.log(randomLat)
+    let randomLong = this.randomLong();
+    console.log(randomLong)
+    let layerLength = leafletPip.pointInLayer(
+      [randomLong, randomLat],
+      this.state.borderData
+    ).length;
+    while (!layerLength) {
+      randomLat = this.randomLat();
+      randomLong = this.randomLong();
+      layerLength = leafletPip.pointInLayer(
+        [randomLong, randomLat],
+        this.state.borderData
+      ).length;
+      this.setState({
+        currentPoint: {lat:randomLat, lng:randomLong}
+      })
+    }
+
+    this.map.panTo([randomLat, randomLong]);
+  }
+
+  // does not function at this time. TODO: activating removes gameStarted state
+  quitGame() {
+    this.setState({ gameStarted: this });
+    if (this.state.gameStarted) {
+      this.map.fitWorld([44.0286, -72.7317], 10);
+    }
+  }
+
+  // checkVermontLocation() {
+  //    { randomLat, randomLong } = this.state;
+  // }
+
+  randomLat() {
+    let lat =
+      Math.random() * (45.00541896831666 - 42.730315121762715) +
+      42.730315121762715;
+    return lat;
+  }
+  randomLong() {
+    let long =
+      Math.random() * (71.51022535353107 - 73.35218221090553) +
+      73.35218221090553 * -1;
+    return long;
+  }
+
+  moveMapNorth() {
```

Comment:
A good first pass at the move methods. Once you finish the backlog you might want to try and refactor these down to a single function since they're all very similar.